### PR TITLE
[Feature] support big query log

### PR DESCRIPTION
**Pull request by silverbullet233**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 29 / 43 (67.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/AuditLogBuilder.java|3|14|21.43%|[94, 95, 96, 97, 98, 107, 109, 111, 113, 115, 117]|
|:large_blue_circle:|com/starrocks/common/AuditLog.java|1|2|50.00%|[41]|
|:large_blue_circle:|com/starrocks/common/Log4jConfig.java|16|18|88.89%|[220, 224]|
|:large_blue_circle:|com/starrocks/common/Config.java|9|9|100.00%|[]|

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 1]
```diff
@@ -157,6 +157,55 @@ public class Config extends ConfigBase {
...
+     * 120s    120 seconds
+     */
+    @ConfField
+    public static String big_query_log_dir = StarRocksFE.STARROCKS_HOME_DIR + "/log";
+    @ConfField
+    public static int big_query_log_roll_num = 10;
+    @ConfField
+    public static boolean enable_big_query_log = false;
+    @ConfField
+    public static long big_query_log_cpu_second_threshold = -1;
```
[the remark this answers is not in the document]
Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java]: > Are session variables better?

fixed

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 2]
```diff
@@ -157,6 +157,55 @@ public class Config extends ConfigBase {
...
+    @ConfField
+    public static long big_query_log_scan_rows_threshold = -1;
```
[the remark this answers is not in the document]
Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java]: > you could reuse some big query session variables

they are now in session variables, the are in resource group's properties. And `big_query_*limit` are hard limits and the query will fail if exceeded. The big query log is just a record and will not interfere with the query execution, so I think it is not suitable to reuse them. so I add some session variables.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 40 / 74 (54.05%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|7|14.29%|[175, 176, 177, 178, 179, 180]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|5|16|31.25%|[1438, 1445, 1446, 1449, 1453, 1454, 1457, 1461, 1462, 1465, 1469]|
|:large_blue_circle:|com/starrocks/plugin/AuditEvent.java|3|9|33.33%|[252, 253, 257, 258, 262, 263]|
|:large_blue_circle:|com/starrocks/common/AuditLog.java|1|2|50.00%|[41]|
|:large_blue_circle:|com/starrocks/qe/AuditLogBuilder.java|9|17|52.94%|[102, 103, 104, 105, 106, 116, 119, 122]|
|:large_blue_circle:|com/starrocks/common/Log4jConfig.java|16|18|88.89%|[220, 224]|
|:large_blue_circle:|com/starrocks/common/Config.java|5|5|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 3]
```diff
@@ -751,6 +756,17 @@ public boolean getUseScanBlockCache() {
     @VarAttr(name = ENABLE_RULE_BASED_MATERIALIZED_VIEW_REWRITE)
     private boolean enableRuleBasedMaterializedViewRewrite = true;
 
+    // if enable_big_query_log = true and cpu/io cost of a query exceeds the related threshold,
+    // the information will be written to the big query log
+    @VarAttr(name = ENABLE_BIG_QUERY_LOG)
+    private boolean enableBigQueryLog = false;
+    @VarAttr(name = BIG_QUERY_LOG_CPU_SECOND_THRESHOLD)
+    private long bigQueryLogCPUSecondThreshold = -1;
```
[the remark this answers is not in the document]
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:764]: +1

[the remark this answers is not in the document]
Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java]: fixed

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 4]
```diff
@@ -751,6 +756,17 @@ public boolean getUseScanBlockCache() {
     @VarAttr(name = ENABLE_RULE_BASED_MATERIALIZED_VIEW_REWRITE)
     private boolean enableRuleBasedMaterializedViewRewrite = true;
 
+    // if enable_big_query_log = true and cpu/io cost of a query exceeds the related threshold,
+    // the information will be written to the big query log
+    @VarAttr(name = ENABLE_BIG_QUERY_LOG)
+    private boolean enableBigQueryLog = false;
```
[the remark this answers is not in the document]
Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java]: fixed

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 55 / 74 (74.32%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|8|16|50.00%|[1447, 1454, 1455, 1462, 1463, 1470, 1471, 1478]|
|:large_blue_circle:|com/starrocks/common/AuditLog.java|1|2|50.00%|[41]|
|:large_blue_circle:|com/starrocks/qe/AuditLogBuilder.java|9|17|52.94%|[102, 103, 104, 105, 106, 116, 119, 122]|
|:large_blue_circle:|com/starrocks/common/Log4jConfig.java|16|18|88.89%|[220, 224]|
|:large_blue_circle:|com/starrocks/plugin/AuditEvent.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|7|7|100.00%|[]|

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Review by murphyatwork — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**